### ISSUE TITLE
Consider using hypothesis for the evaluation metrics tests

### ISSUE THREAD

**Issue by fluegelk** [Helmholtz-AI-Energy/special-couscous#17]
[`hypothesis`](https://hypothesis.readthedocs.io/en/latest/quickstart.html) seems like a great fit for the evaluation metric tests as all metrics can be compared against sklearn's implementation from sample-wise inputs, i.e., the following assumption should hold:
```
sklearn.metric.<metric>(y_true, y_pred) == evaluation_metrics.<metric>(sklearn.metric.confusion_matrix(y_true, y_pred))
```
